scalameta/io/shared/src/main/scala/scala/meta/io/RelativePath.scala

```diff
@@ -6,12 +6,26 @@ import java.nio.file.Path
 import java.nio.{file => nio}
 import java.nio.file.Paths
 import scala.meta.internal.io.PathIO
+import scala.collection.JavaConverters._
 
 /** Wrapper around a relative nio.Path. */
 sealed abstract case class RelativePath(toNIO: Path) {
   require(!toNIO.isAbsolute, s"$toNIO is not relative!")
   def toFile: File = toNIO.toFile
-  def toURI: URI = toFile.toURI
+  @deprecated("Use toURI(isDirectory = false) instead", "4.0.0")
+  def toURI: URI = {
```

Comment:
I'm not sure whether it's worth keeping this method around. As far as I can see, it has potential to produce incorrect results, so I think that breaking binary compatibility is the right thing to do here. Compatibility with incorrect behavior doesn't sound like a good idea.